smac_planner/src/node_se2.cpp

```diff
@@ -232,11 +243,24 @@ bool NodeSE2::isNodeValid(const bool & traverse_unknown)
 
 float NodeSE2::getHeuristicCost(
   const Coordinates & node_coords,
-  const Coordinates & goal_coordinates)
+  const Coordinates & goal_coords)
 {
-  return hypotf(
-    goal_coordinates.x - node_coords.x,
-    goal_coordinates.y - node_coords.y);
+  if (!_motion_model.state_space) {
```

Comment:
I think what we should be doing is computing both and using the highest, no? 

I don't think by this logic cartesian would ever be used. All motion models will have a `state_space`.

Reply:
If every motion model is guaranteed to have a `state_space` then I agree. I put it there in case someone came up with a motion model that didn't have a corresponding OMPL state space.
I will remove.
Also, the highest is guaranteed to be the one calculated on the state space - best case is that it is a straight line (Cartesian equivalent), but most times it will have curves too.